### PR TITLE
feat: add names

### LINKED ISSUE #12: show bns names as username

### PR DESCRIPTION
**Pull request by fbwoolf**

### CODE REVIEW

File: src/common/hooks/use-fetch.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const hiroHeaders: HeadersInit = {
+  'x-hiro-product': 'heystack',
+  'x-hiro-version': '1.0.0', // TODO: Global VERSION
```
Comment (kyranjamie) [inline src/common/hooks/use-fetch.ts]: Can pull in direct from package.json?

File: src/store/hey.ts [hunk 2]
```diff
@@ -75,7 +75,7 @@ export const pendingTxsAtom = atomWithQuery<Heystack[], string>(get => ({
       )
       .map(tx => tx.tx_id);
     const final = await Promise.all(
-      heyTxs.map(async txid => client.getTransactionById({ txId: txid }))
+      heyTxs.map(async txid => await client.getTransactionById({ txId: txid }))
```
Comment (kyranjamie) [inline src/store/hey.ts]: Not necessary when constructing a Promise.all
```suggestion
      heyTxs.map(txid => client.getTransactionById({ txId: txid }))
```

File: src/components/auth.tsx [hunk 3]
```diff
@@ -22,12 +23,13 @@ import { UserAvatar } from '@components/user-avatar';
 const Dropdown: React.FC<BoxProps & { onSignOut?: () => void; show?: boolean }> = memo(
   ({ onSignOut, show }) => {
     return (
-      <Fade in={show}>
+      <Fade in={true}>
```
Comment (kyranjamie) [inline src/components/auth.tsx]: 🤔 ?

File: src/store/names.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+function makeKey(networkUrl: string, address: string): string {
+  return `${networkUrl}__${address}`;
+}
+
+function getLocalNames(networkUrl: string, address: string): [string[], number] | null {
+  const key = makeKey(networkUrl, address);
+  const value = localStorage.getItem(key);
+  if (!value) return null;
+  return JSON.parse(value);
```
Comment (kyranjamie) [inline src/store/names.ts:22]: JSON.parse with invalid json throws an error. Generally best to use a wrapper lib when dealing with localStorage, which contains try/catch logic, and handling of things like `'true'` --> `true`

File: src/store/hey.ts (hunk 2 above)
Comment (fbwoolf) [reply src/store/hey.ts]: My linter forced me to add the `await` when I rebased but overlooked just removing the `async`, thx!

File: src/components/auth.tsx (hunk 3 above)
Comment (fbwoolf) [reply src/components/auth.tsx]: O, that was just for testing! Sorry, will put back.

File: src/common/hooks/use-fetch.ts (hunk 1 above)
Comment (fbwoolf) [reply src/common/hooks/use-fetch.ts]: I ported this over from the web wallet, but maybe don't need.

File: src/components/auth.tsx [hunk 5]
```diff
@@ -56,9 +62,16 @@ const Dropdown: React.FC<BoxProps & { onSignOut?: () => void; show?: boolean }>
     );
   }
 );
+
+const AccountNameComponent = memo(() => {
+  const address = useCurrentAddress();
+  const accountName = useAccountName(address);
+  return <Text>{accountName?.names}</Text>;
```
Comment (fbwoolf) [inline src/components/auth.tsx]: This is an empty array atm, but not sure why yet.

File: src/store/names.ts (hunk 4 above)
Comment (fbwoolf) [reply src/store/names.ts:22]: I see now there is an `atomWithStorage` to use so will try to refactor.

File: src/components/auth.tsx (hunk 5 above)
Comment (aulneau) [reply src/components/auth.tsx]: BNS names are network specific. meaning if you have fara.btc on mainnet, it won't show up on testnet

Comment (fbwoolf) [reply src/components/auth.tsx]: Ok, got it, so prob won't display a BNS name very often so would just show if there is one. I am a little confused about why this is an array if a BNS name is tied to one account?

Comment (aulneau) [reply src/components/auth.tsx]: You can have many off chain names, eg fara.stx.id

**Pull request merged**